ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -804,22 +804,13 @@ generateGhcLibCabal ghcFlavor = do
     putStrLn "# Generating 'ghc-lib.cabal'... Done!"
 
 ghciDef :: GhcFlavor -> String
-ghciDef GhcMaster = ""
-ghciDef Ghc901 = ""
-ghciDef Ghc8101 = ""
-ghciDef Ghc8102 = ""
-ghciDef Ghc8103 = ""
-ghciDef Ghc8104 = ""
-ghciDef _ = "-DGHCI"
+ghciDef ghcFlavor = if ghcFlavor > Ghc8101 then "" else "-DGHCI"
 
 ghcStageDef :: GhcFlavor -> String
-ghcStageDef GhcMaster = ""
-ghcStageDef Ghc901 = ""
-ghcStageDef Ghc8101 = ""
-ghcStageDef Ghc8102 = ""
-ghcStageDef Ghc8103 = ""
-ghcStageDef Ghc8104 = ""
-ghcStageDef _ = "-DSTAGE=2"
+ghcStageDef ghcFlavor = if ghcFlavor >= Ghc8101 then "" else "-DSTAGE=2"
+
+ghcInGhciDef :: GhcFlavor -> String
```

Comment:
let’s leave it out of this PR but returning empty strings seems a bit of an odd choice. We should return a `Maybe String` or a `[String]` and then throw `unwords` at it at the call site.

Reply:
Issue: https://github.com/digital-asset/ghc-lib/issues/274